tests/tracing/test_otel_logging.py

```diff
@@ -336,3 +336,261 @@ def test_empty_resource_spans_returns_400(mlflow_server: str):
+    response = requests.post(
+        f"{mlflow_server}/v1/traces",
+        data=request.SerializeToString(),
+        headers={
+            "Content-Type": "application/x-protobuf",
+            MLFLOW_EXPERIMENT_ID_HEADER: experiment_id,
+        },
+        timeout=30,
+    )
+    elapsed_time = time.time() - start_time
+
+    assert response.status_code == 200, f"Expected 200, got {response.status_code}"
+    # With parallel processing, this should complete reasonably fast
+    # (exact timing depends on hardware, but should be under 10 seconds)
+    assert elapsed_time < 10, f"Request took {elapsed_time}s, expected < 10s"
+
+    # Wait for all traces to be available
+    time.sleep(2)
+
+    traces = mlflow.search_traces(
+        experiment_ids=[experiment_id], include_spans=False, return_type="list"
+    )
+
+    assert len(traces) == num_traces, f"Expected {num_traces} traces, found {len(traces)}"
+
+
+def test_mixed_trace_spans_in_single_request(mlflow_server: str):
+    """
+    Test that multiple spans from the same trace, mixed with spans from other traces,
+    are grouped and logged correctly.
+    """
+    mlflow.set_tracking_uri(mlflow_server)
+    experiment = mlflow.set_experiment("otel-mixed-test")
+    experiment_id = experiment.experiment_id
+
+    request = ExportTraceServiceRequest()
+
+    # Create 2 spans for trace A, 1 span for trace B, 2 spans for trace C
+    trace_configs = [
+        ("A", 0, 2),  # trace A with 2 spans
+        ("B", 1, 1),  # trace B with 1 span
+        ("C", 2, 2),  # trace C with 2 spans
+    ]
+
+    for trace_name, trace_id_num, span_count in trace_configs:
+        trace_id_hex = f"{trace_id_num + 2000:016x}" + "0" * 16
+
+        for span_num in range(span_count):
+            span = OTelProtoSpan()
+            span.trace_id = bytes.fromhex(trace_id_hex)
+            span.span_id = bytes.fromhex(f"{trace_id_num * 100 + span_num:08x}" + "0" * 8)
+            span.name = f"mixed-span-{trace_name}-{span_num}"
+            span.start_time_unix_nano = 1000000000 + span_num * 1000
+            span.end_time_unix_nano = 2000000000 + span_num * 1000
+
+            scope = InstrumentationScope()
+            scope.name = "mixed-test-scope"
+
+            scope_spans = ScopeSpans()
+            scope_spans.scope.CopyFrom(scope)
+            scope_spans.spans.append(span)
+
+            resource = Resource()
+            resource_spans = ResourceSpans()
+            resource_spans.resource.CopyFrom(resource)
+            resource_spans.scope_spans.append(scope_spans)
+
+            request.resource_spans.append(resource_spans)
+
+    response = requests.post(
+        f"{mlflow_server}/v1/traces",
+        data=request.SerializeToString(),
+        headers={
+            "Content-Type": "application/x-protobuf",
+            MLFLOW_EXPERIMENT_ID_HEADER: experiment_id,
+        },
+        timeout=10,
+    )
+
+    assert response.status_code == 200
+
+    # Wait for traces
+    time.sleep(1)
```

Comment:
Why do we need to sleep? Is span logging async?

Reply:
It's not, right? We call `log_spans` and wait for it to complete, and `requests` is sync.